content/raw/wallet/add_account.md

```diff
@@ -0,0 +1,123 @@
+
+## Motivation
+
+Users often want to add an account to their wallet and this is a very common use case. These accounts can either be generated, or imported. In some cases Users also want to add addresses just to watch them.
+
+## Definitions
+
+- ethereum address - hexadecimal string of 64 characters used to identify an account
+- private key - hexadecimal string of 64 characters that is used to sign transactions belonging to the account
+- seed phrase - a list of 12 words used to generate one or more private keys
+- derivation path - a list of integers used to derive a private key from a seed phrase 
+- ENS - a ethereum domain name that is used to resolve an address
+- assets - refers to Ethereum Tokens and ERC-20 Tokens
+- active account - an ethereum address that has transactions
+
+## Designs
+
+- [designs](https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=49%3A1737962)
+
+## Use Cases
+
+### Generate an account
+
+**TODO** design is missing
+
+1. User selects "Generate an account"
+2. System shows dialog to enter account name and emoji
+3. System randomly selects an emoji
+4. User fills in account name
+5. (optional): User selects emoji
+6. User selects "Add account"
+
+### Import an account from a private key
+
+1. User selects "Add with a key or seed phrase"
+2. System shows dialog to enter a private key or seed phrase, name and emoji
+3. System randomly selects an emoji
+4. User enters private key
+4.1 alternative: user fills seed phrase, see "Import an account from a seed phrase"
+5. Systems validates private key
+6. User fills name
+7. (optional) User selects emoji
+8. User selects "Add account"
+
+### Import an account from a seed phrase
+
+1. User selects "Add with a key or seed phrase"
+2. System shows dialog to enter a private key or seed phrase, name and emoji
+3. System randomly selects an emoji
+4. User enters seed phrase
+5.1 alternative: user fills private key, see "Import an account from a private key"
+5. System searches for "active accounts"
+6. System displays list of known active accounts
+6.1 alternative: System displays "We found no active accounts with that seed phrase"
+6.1.1 System offers option to generate an account, see "Generate an account
+6.1.1 System offers option to proceed with seed phrase, see "Import an account from a seed phrase: no active accounts found or new derivation path"
+7. (optional) User deletes one or more "active account"
+8. User fills names of active accounts
+9. (optional) User changes emoji of an active account
+10 User selects "Add accounts"
+10.1 alternative: user selects: "Add new derivation path", see "Import an account from a seed phrase: no active accounts found or new derivation path"
+
+#### Import an account from a seed phrase: no active accounts found or new derivation path
+
+1. System displays a list of derivation paths
+2. User selects one of the derivation paths
+2.1 alternative: User types derivation path
+3. **TODO** design is missing next steps
```

Comment:
the next steps here seem to be missing, for example typically multiples accounts are shown to the user that match the derivation path and the user selects one or some of them. unless I missunderstood and this is suppose to add one account at the time? (i.e the derivation path needs to include the exact index)